mongosql/src/algebrizer/test.rs

```diff
@@ -7454,20 +7454,256 @@ mod limit_or_offset_clause {
 }
 
 mod set_query {
-    use super::{catalog, mir_source_bar, mir_source_foo, AST_QUERY_BAR, AST_QUERY_FOO};
-    use crate::{ast, mir, mir::schema::SchemaCache, usererror::UserError};
+    use super::{
+        catalog, mir_source_bar, mir_source_foo, AST_QUERY_BAR, AST_QUERY_FOO, AST_SOURCE_BAR,
+        AST_SOURCE_FOO,
+    };
+    use crate::schema::{Document, Schema};
+    use crate::{
+        ast, map, mir, mir::schema::SchemaCache, multimap, schema, set,
+        unchecked_unique_linked_hash_map,
+    };
+    use mongosql_datastructures::binding_tuple::DatasourceName;
 
     test_algebrize!(
-        union_distinct_not_allowed,
+        union_distinct_star,
         method = algebrize_set_query,
-        expected = Err(Error::DistinctUnion),
-        expected_error_code = 3006,
+        expected = Ok(mir::Stage::Project(mir::Project {
+            source: Box::new(mir::Stage::Group(mir::Group {
+                source: Box::new(mir::Stage::Set(mir::Set {
+                    operation: mir::SetOperation::UnionAll,
+                    left: Box::new(mir::Stage::Project(mir::Project {
+                        source: Box::new(mir::Stage::Collection(mir::Collection {
+                            db: "test".into(),
+                            collection: "foo".into(),
+                            cache: SchemaCache::new()
+                        })),
+                        expression: map! {
+                            (DatasourceName::Named("foo".into()), 0u16).into() => mir::Expression::Reference((DatasourceName::Named("foo".into()), 0u16).into())
+                        },
+                        is_add_fields: false,
+                        cache: SchemaCache::new()
+                    })),
+                    right: Box::new(mir::Stage::Project(mir::Project {
+                        source: Box::new(mir::Stage::Collection(mir::Collection {
+                            db: "test".into(),
+                            collection: "bar".into(),
+                            cache: SchemaCache::new()
+                        })),
+                        expression: map! {
+                            (DatasourceName::Named("bar".into()), 0u16).into() => mir::Expression::Reference((DatasourceName::Named("bar".into()), 0u16).into())
+                        },
+                        is_add_fields: false,
+                        cache: SchemaCache::new()
+                    })),
```

Comment:
```suggestion
                    left: Box::new(mir_source_foo()),
                    right: Box::new(mir_source_bar()),
```
We could reuse some of the global helpers here to reduce the size of these tests! We could even go a step further and simplify these two helpers into one that is parametrized by collection name. Up to you if you want to do that kind of refactoring here or not.

Reply:
I added a parameterized helper function here.